### ISSUE TITLE
Add option to disable `lake build`

### ISSUE THREAD

**Issue by Seasawher** [leanprover/lean-action#30]
I sometimes want just to install `elan` and not to run `lake build`

**Comment by oliver-butterley** [comment]
In all the use cases I have encountered it works well to simply include the step:
```yml
      - name: Install elan
        run: |
          curl https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh -sSf | sh -s -- -y
          echo "$HOME/.elan/bin" >> $GITHUB_PATH
```
Do you have something in mind where it would be better to use `lean-action` with extra options for the task?

**Comment by Seasawher** [comment]
It is an advantage to be able to provide standard commands to install elan.
After `lake build` has been replaced by lean-action, it may not be clear what the standard commands are for installing elan.

**Comment by Seasawher** [comment]
For example, I often see elan installed with commands such as

```yaml
          run: |
            curl -sSfL https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh | sh -s -- -y --default-toolchain none
            echo "$USERPROFILE\.elan\bin" >> $GITHUB_PATH
```

**Comment by oliver-butterley** [comment]
- It does have the advantage that if the way to install elan changes then only `lean-action` needs updating, all other workflows which use it remain as before. 

- Additionally, gh action caching is built into `lean-action` and so could make workflows that use it faster without them needing to add the extra lines to setup caching.

**Comment by oliver-butterley** [comment]
> For example, I often see elan installed with commands such as
> 
> ```yaml
>           run: |
>             curl -sSfL https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh | sh -s -- -y --default-toolchain none
>             echo "$USERPROFILE\.elan\bin" >> $GITHUB_PATH
> ```

Documentation for the possible arguments for `elan-init.sh` is a bit hidden, I've only seen it in the [script file](https://github.com/leanprover/elan/blob/master/elan-init.sh). Different options here to fit different purposes.

Concerning the path, there are various ways that work, the prior one is the [suggested way](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path).

**Comment by austinletson** [comment]
As we expand the inputs to support more use cases, we may want to consider changing the behavior of the inputs so that all the steps are disabled by default and you specify what you want when you call the action. This style is inspired by [python-action](https://github.com/microsoft/action-python).

For example,
```yaml
- uses: leanprover/lean-action
  with:
    build: true
    test: true
    lint: true
```
The current behavior is a set of steps which are enabled by default (`lake build` and `lake test`) and a set which are disabled by default (linting, checking reservoir eligibility, and `lean4checker`). We then expect the user to know which of the first set they wish to disable and which of the second set they wish to enable. It may be simpler to give the user a menu of steps which they can enable and let them choose what they want.

Note we could still provide a sensible default configuration either when the action is called with no inputs or with a `default` input:
```yaml
- uses: leanprover/lean-action
  with:
    default: true
```

**Comment by austinletson** [comment]
I ran into a use case for this today while adding some automated functional testing. [link](https://github.com/leanprover/lean-action/pull/35/files#diff-40b9c18b065a9e9abf792b0b2fb79e10299ad9bffb91dd2c68a370779f9001cbR23)

**Comment by austinletson** [comment]
Closed by #61
